python/pylibvw.cc

```diff
@@ -542,32 +542,85 @@ std::string varray_char_to_string(v_array<char>& a)
 
 std::string my_get_tag(example_ptr ec) { return varray_char_to_string(ec->tag); }
 
-uint32_t ex_num_namespaces(example_ptr ec) { return (uint32_t)ec->indices.size(); }
+uint32_t ex_num_namespaces(example_ptr ec) { return (uint32_t)ec->feature_space.size(); }
 
-unsigned char ex_namespace(example_ptr ec, uint32_t ns) { return ec->indices[ns]; }
+unsigned char ex_namespace(example_ptr ec, uint32_t distance) 
+{
+  auto it = ec->begin();
+  for (int i = 0; i < distance; ++i, ++it);
+  return it.index();
+}
 
-uint32_t ex_num_features(example_ptr ec, unsigned char ns) { return (uint32_t)ec->feature_space[ns].size(); }
+uint32_t ex_num_features(example_ptr ec, char* ns) {
+  auto fs = ec->feature_space;
+  uint64_t hash = hashstring(ns, strlen(ns), 0);
+  return (strlen(ns) == 1) ? fs.namespace_index_end_proxy(ns[0]) - fs.namespace_index_begin_proxy(ns[0]) : fs[hash].size();
```

Comment:
It is still valid to have a namespace of length 1 and access it using the hash so we need to make sure it works with that. So it may best to use an explicit bool parameter for switching behavior

Reply:
I changed the logic in these cases to check if the length is 1, and if the hash does not exist